tests/ccl_test_core.py

```diff
@@ -303,5 +303,21 @@ def test_cosmology_repr():
         ccl.comoving_radial_distance(cosmo3, 0.5))
 
 
+def test_cosmology_context():
+    with ccl.Cosmology(
```

Comment:
Could do with docstring explaining what this test does

Reply:
Done.